pontoon/sync/tests/test_changeset.py

```diff
@@ -300,3 +301,41 @@ def test_obsolete_db(self):
         self.changeset.execute()
         self.main_db_entity.refresh_from_db()
         assert_true(self.main_db_entity.obsolete)
+
+    def test_multiple_locale_authors(self):
+        """
+        Test if authors are allocated per locale properly.
+        Regression tests against bug 1214722.
+
+        Contributors should map to locales following way:
+            first_locale -> first_user, third_user
+            second_locale -> second_user
+        """
+        first_entity, second_entity = EntityFactory.create_batch(2)
+        first_locale, second_locale = LocaleFactory.create_batch(2)
+        first_author, second_author, third_author = UserFactory.create_batch(3)
+
+        TranslationFactory.create(locale=first_locale, user=first_author, entity=first_entity)
+        TranslationFactory.create(locale=second_locale, user=second_author, entity=first_entity)
+        TranslationFactory.create(locale=first_locale, user=third_author, entity=first_entity)
+        # Because of logic contained in Translation.save() we have too update
+        # approved field here.
+        Translation.objects.update(approved=True)
+
+        self.vcs_project.resources[first_entity.resource.path] = first_entity.resource
+        self.vcs_project.resources[second_entity.resource.path] = second_entity.resource
+
+        self.changeset.changes['update_vcs'] = (
```

Comment:
You should use `self.changeset.update_vcs_entity` instead of directly assigning to this list, it's an internal detail about changesets you should avoid relying on.

Also you should avoid using mocks for the VCS entities if you can avoid it, which is another reason to use the already-set-up entities instead of creating new ones that don't exist in the sample files.